src/auto/pollable_input_stream.rs

```diff
@@ -0,0 +1,60 @@
+// This file was generated by gir (5a68ad0) from gir-files (469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+use InputStream;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct PollableInputStream(Object<ffi::GPollableInputStream, ffi::GPollableInputStreamInterface>): InputStream;
+
+    match fn {
+        get_type => || ffi::g_pollable_input_stream_get_type(),
+    }
+}
+
+pub trait PollableInputStreamExt {
+    fn can_poll(&self) -> bool;
+
+    //fn create_source<'a, P: Into<Option<&'a Cancellable>>>(&self, cancellable: P) -> /*Ignored*/Option<glib::Source>;
+
+    fn is_readable(&self) -> bool;
+
+    fn read_nonblocking<'a, P: Into<Option<&'a Cancellable>>>(&self, buffer: &[u8], cancellable: P) -> Result<isize, Error>;
```

Comment:
This should probably be manual with an `AsMut<[u8]>` (and is currently wrong: it has to be at least `&mut [u8]`).